src/core/colorspace.js

```diff
@@ -379,24 +379,46 @@ class AlternateCS extends ColorSpace {
       : new Uint8ClampedArray(baseNumComps * count);
     const numComps = this.numComps;
 
+    const tintCache = new Map();
+    // Use an integer cache-key when possible, since that's a lot faster than
+    // string concatenation.
+    const int32Key =
+      numComps <= 3 &&
```

Comment:
Why 3 and not 4? If we have four 8-bit elements, we should be able to fit all of them in a int32 (and sometimes we'll get negative number, if the high bit is set).

Note that I might be wrong here, but I tried with a bunch of numbers and it seems like it's possible to extract them back (so there is a 1-1 relationship between the 4 bytes and the int32, when mangling them with <<):
```
(200 << 24) + (199 << 16) + (198 << 8) + 197; // -926431547
(((200 << 24) + (199 << 16) + (198 << 8) + 197) >>> 0) & 255; // 197
(((200 << 24) + (199 << 16) + (198 << 8) + 197) >>> 8) & 255; // 198
(((200 << 24) + (199 << 16) + (198 << 8) + 197) >>> 16) & 255; // 199
(((200 << 24) + (199 << 16) + (198 << 8) + 197) >>> 24) & 255; // 200
```

Reply:
> if it's somehow possible for the addition of a negative number to produce "clashing" cache-keys?

It cannot because the distance between two consecutive negative numbers obtained by `x << 24` is always `16777216`, which is larger than `255 + (255 << 8) + (255 << 16)` (16777215).

It seems like, at least on little-endian machines, `key = src[0] + (src[1] << 8) + (src[2] << 16) + (src[3] << 24)` is the same as doing `key = new Int32Array(src.buffer)[0]`, which obviously gives different results for different 4 bytes :)

Reply:
I checked various values "around" the point where the int32 becomes negative, and it seems fine; thank you.